geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStore.scala

```diff
@@ -135,42 +59,55 @@ object KafkaDataStoreFactoryParams {
-  val EXPIRY             = new Param("expiry", classOf[java.lang.Boolean], "Expiry", false, false)
-  val EXPIRATION_PERIOD  = new Param("expirationPeriod", classOf[java.lang.Long], "Expiration Period in milliseconds", false, false)
 }
 
+object KafkaDataStore {
+  type FeatureSourceFactory = (ContentEntry, KafkaDataStoreSchemaManager) => ContentFeatureSource
+}
+
+/** A [[DataStoreFactorySpi]] to create a [[KafkaDataStore]] in either producer or consumer mode */
 class KafkaDataStoreFactory extends DataStoreFactorySpi {
 
+  import org.locationtech.geomesa.kafka.KafkaDataStore._
   import org.locationtech.geomesa.kafka.KafkaDataStoreFactoryParams._
 
   override def createDataStore(params: ju.Map[String, Serializable]): DataStore = {
     val broker   = KAFKA_BROKER_PARAM.lookUp(params).asInstanceOf[String]
     val zk       = ZOOKEEPERS_PARAM.lookUp(params).asInstanceOf[String]
-    val zkPath   = Option(ZK_PATH.lookUp(params).asInstanceOf[String])
-                     .map(_.trim)
-                     .filterNot(_.isEmpty)
-                     .map(p => if (p.startsWith("/")) p else "/" + p)
-                     .map(p => if (p.endsWith("/")) p.substring(0, p.length - 1) else p)
-                     .getOrElse("/geomesa/ds/kafka")
-    val isProducer =
-      if(IS_PRODUCER_PARAM.lookUp(params) == null) java.lang.Boolean.FALSE
-      else IS_PRODUCER_PARAM.lookUp(params).asInstanceOf[java.lang.Boolean]
+    val zkPath   = KafkaDataStoreHelper.cleanZkPath(ZK_PATH.lookUp(params).asInstanceOf[String])
 
     val partitions       = Option(TOPIC_PARTITIONS.lookUp(params)).map(_.toString.toInt).getOrElse(1)
     val replication      = Option(TOPIC_REPLICATION.lookUp(params)).map(_.toString.toInt).getOrElse(1)
-    val expiry           = Option(EXPIRY.lookUp(params).asInstanceOf[java.lang.Boolean]).getOrElse(java.lang.Boolean.FALSE)
-    val expirationPeriod = Option(EXPIRATION_PERIOD.lookUp(params)).map(_.toString.toLong).getOrElse(0L)
 
-    new KafkaDataStore(broker, zk, zkPath, partitions, replication, isProducer, expiry, expirationPeriod)
+    val fsFactory = createFeatureSourceFactory(broker, zk, params)
+
+    new KafkaDataStore(zk, zkPath, partitions, replication, fsFactory)
+  }
+
+  def createFeatureSourceFactory(brokers: String,
+                                 zk: String,
+                                 params: ju.Map[String, Serializable]): FeatureSourceFactory = {
+
+    val isProducer = Option(IS_PRODUCER_PARAM.lookUp(params).asInstanceOf[Boolean]).getOrElse(false)
+
+    if (isProducer) {
+      KafkaProducerFeatureStoreFactory(brokers)
+    } else {
+      KafkaConsumerFeatureSourceFactory(brokers, zk, params)
+    }
   }
 
-  override def createNewDataStore(params: ju.Map[String, Serializable]): DataStore = ???
+  override def createNewDataStore(params: ju.Map[String, Serializable]): DataStore =
+    throw new UnsupportedOperationException
```

Comment:
Hmm... did the compiler make you add this?  I imagine the override would have taken care of the exception declaration.  

I think it'll be fine either way.